### PR TITLE
doc/user: improve documentation on `SHOW CLUSTERS`

### PR DESCRIPTION
**Pull request by morsapaes**
Address feedback from @heeringa. This commit makes the page more consistent, and adds additional information around the purpose of the `mz_system` system cluster.

### CODE REVIEW

File: doc/user/content/sql/show-clusters.md [hunk 1]
```diff
@@ -17,53 +17,55 @@ menu:
...
-```
+  * You can run `SELECT` or `SUBSCRIBE` statements on this cluster.
 
 ### `mz_system` system cluster
 
-The `mz_system` cluster is used for various internal system tasks. You are not
-billed for this cluster. You are not permitted to use this cluster.
-Specifically:
+A system cluster named `mz_system` will be pre-installed in every environment.
+This cluster is used for different system administration tasks, like billing
```
Comment (benesch) [inline doc/user/content/sql/show-clusters.md:54]: "different system administration tasks" here raises the question of "different from what?" I'd also prefer to be somewhat circumspect with our description of its purpose; the idea is that it's running who knows what to support the system and you, as the end user, shouldn't know or care. What do you think about saying just "This cluster is used by the system for various internal monitoring tasks", or something like that?

Comment (morsapaes) [reply doc/user/content/sql/show-clusters.md:54]: Yeah, I went back and forth on it; especially about mentioning billing. @heeringa asked if we could be clearer about the purpose, but...maybe we just don't need to be.

Comment (heeringa) [reply doc/user/content/sql/show-clusters.md]: LGTM

**Review by heeringa — APPROVED** [review]
LGTM

**Review by benesch — APPROVED** [review]

**Pull request merged**